pkg/security/podsecuritypolicy/capabilities/mustrunas_test.go

```diff
@@ -308,36 +298,35 @@ func TestValidateAdds(t *testing.T) {
 			},
 		}
 
-		strategy, err := NewDefaultCapabilities(v.defaultAddCaps, v.requiredDropCaps, v.allowedCaps)
+		strategy, err := NewDefaultCapabilities(v.defaultAddCaps, nil, v.allowedCaps)
 		if err != nil {
 			t.Errorf("%s failed: %v", k, err)
 			continue
 		}
 		errs := strategy.Validate(nil, container)
-		if v.shouldPass && len(errs) > 0 {
-			t.Errorf("%s should have passed but had errors %v", k, errs)
+		if len(v.expectedErrors) != len(errs) {
+			t.Errorf("Expected %v (%v) errors, got %v (%v)", len(v.expectedErrors), v.expectedErrors, len(errs), errs)
 			continue
 		}
-		if !v.shouldPass && len(errs) == 0 {
-			t.Errorf("%s should have failed but received no errors", k)
+		for i := 0; i < len(v.expectedErrors); i++ {
```

Comment:
- Made it less general
- Then I also stopped creating errors and just came back to strings